Core/HLE/sceKernelHeap.cpp

```diff
@@ -110,18 +110,65 @@ static u32 SysMemForKernel_536AD5E1()
 	return 0;
 }
 
+static int sceKernelFreeHeapMemory(int heapId, u32 block) {
+	u32 error;
+	KernelHeap* heap = kernelObjects.Get<KernelHeap>(heapId, error);
+	if (heap) {
+		if (block == 0)
+			INFO_LOG(SCEKERNEL, "sceKernelFreeHeapMemory(%d): heapId, block", heapId, 0);
+			return 0;
+		if (!heap->alloc.FreeExact(block)) {
+			ERROR_LOG(SCEKERNEL, "sceKernelFreeHeapMemory(%d): heapId, block Invalid pointer ", heapId, block);
+			return SCE_KERNEL_ERROR_INVALID_POINTER;
+		}
+		return hleLogSuccessInfoI(SCEKERNEL, 0, "sceKernelFreeHeapMemory(%d): heapId, block", heapId, block);;
+	}
+	else {
+		return hleLogError(SCEKERNEL, error, "sceKernelFreeHeapMemory(%d): invalid heapId", heapId);
+	}
```

Comment:
It's better to replace:

```c++
if (blah) {
    return a;
} else {
    return b;
}
```

With:

```c++
if (blah) {
    return a;
}

return b;
```

Although, in this case it'd probably be even better to use:

```c++
if (!heap)
	return hleLogError(SCEKERNEL, error, "sceKernelFreeHeapMemory(%d): invalid heapId", heapId);

if (block == 0) {
...
```

-[Unknown]